macros/src/test/scala/gestalt/macros/DefMacroTest.scala

```diff
@@ -211,4 +211,9 @@ class DefMacroTest extends TestSuite {
     assert(trees.some3Explicit() == Some(3))
     assert(trees.some3ExplicitTyped() == Some(3))
   }
+
+  test("white box") {
+    val x = Whitebox.gimmelist(2)
```

Comment:
With/without explicit type annotation, the type checking go through different steps. If without, `x` gets a type from the rhs, which is expanded and type checked as `List[Int]`.

With expected type `List[Int]`, there is an assert before macro expansion, thus compilation failed. We could change the implementation to try to expand macros before fail with expected type.

Anyway, I had the impression that whitebox doesn't have much to do with macro system design. It's more a language design decision.